File: Add Code Here/C++/Majority element II.cpp
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
        unordered_map<int,int> m;
        for(int i=0;i<nums.size();i++){
            m[nums[i]]++;
            
        }
        int n=nums.size();
        vector<int> ans;
        for(auto x:m){
            if(x.second>n/3){
                ans.push_back(x.first);
            }
        }
        return ans;
        
        
    }
};

File: Add Code Here/JAVA/Java_Medium_Solutions/MajorityElement-II.java
class Solution {
    public List<Integer> majorityElement(int[] nums) {
        List<Integer> ans=new ArrayList<>();
        HashMap<Integer,Integer> map=new HashMap<>();
        int n=nums.length;
        for(int ele: nums) {
            map.put(ele, map.getOrDefault(ele,0)+1);
            if(map.containsKey(ele)) {
                int count=map.get(ele);
                if(count>(n/3) && !ans.contains(ele)) {
                    ans.add(ele);
                    map.remove(ele);
                }
            }
        }
        return ans;
    }
}